### PR TITLE
Spidermonkey 60 bump

### PR DESCRIPTION
**Pull request by hedning**
###### Motivation for this change

Bump spidermonkey 60, and drop unused 52.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/interpreters/spidermonkey/60.nix [hunk 1]
```diff
@@ -4,23 +4,26 @@
...
   };
 
+  outputs = [ "out" "dev" ];
+  setOutputFlags = false; # Configure script only understands --includedir
+
   buildInputs = [ readline zlib icu ];
   nativeBuildInputs = [ autoconf213 pkgconfig perl which python2 zip ];
 
   patches = [
     (fetchpatch {
```
Comment (jtojnar) [inline pkgs/development/interpreters/spidermonkey/60.nix]:
```suggestion
    # Fixed in 62.0
    # https://bugzilla.mozilla.org/show_bug.cgi?id=1415202
    (fetchpatch {
```

Comment (hedning) [reply pkgs/development/interpreters/spidermonkey/60.nix]: cheers

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -9034,7 +9034,6 @@ in
   } // (stdenv.lib.optionalAttrs (stdenv.cc.isGNU && stdenv.hostPlatform.isi686) {
       stdenv = gcc6Stdenv; # with gcc-7: undefined reference to `__divmoddi4'
   }));
-  spidermonkey_52 = callPackage ../development/interpreters/spidermonkey/52.nix { };
```
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]: Do we want aliases that throw to use the latest attribute?

Comment (hedning) [reply pkgs/top-level/all-packages.nix]: I guess, no idea if this was a common package for users to install/use.

File: pkgs/top-level/aliases.nix [hunk 3]
```diff
@@ -344,6 +344,7 @@ mapAliases ({
   speedtest_cli = speedtest-cli;  # added 2015-02-17
   spice_gtk = spice-gtk; # added 2018-02-25
   spice_protocol = spice-protocol; # added 2018-02-25
+  spidermonkey_52 = throw "deprecrated 2019-10-16: use spidermonkey_60 instead";
```
Comment (worldofpeace) [inline pkgs/top-level/aliases.nix]: I'm not sure I like these messages formatted like this.
It's confusing to people, was `spidermonkey_52` deprecated that date or did nixpkgs do this?

```suggestion
  spidermonkey_52 = throw "spidermonkey_52 has been removed. Please use spidermonkey_60 instead."; # added 2019-10-16
```

**Comment by worldofpeace** [comment]
@GrahamcOfBorg eval

**Comment by worldofpeace** [comment]
@GrahamcOfBorg eval

**Pull request merged**